File: Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Meal.Models
{
    public class Order
    {
        [Key] public int Id { get; set; }
        [Required] public string UserId { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; }
        public DateTime? Date { get; set; }
        public User User { get; set; }
    }
}

File: src/meal/ClientApp/src/app/models/order.ts
import {User} from "./user";

export interface Order {
    id: number;
    userId: string;
    user: User;
    orderItems: OrderItem[]
}

export interface OrderItem {
    id: number
    mealType: MealType,
    orderId: number;
    name: string;
}

export interface Meal {
    id: number,
    name: string;
    mealType: MealType;
}

export enum MealType {
    Mains,
    SideOrders,
    Salad
}
